### PR TITLE
[MM-42660] Create a timeline event when an update is snoozed.

### PR DESCRIPTION
**Pull request by jeffreyssmith2nd**
#### Summary

Adds a timeline event to record when a status update reminder is snoozed.

#### Ticket Link

Fixes: https://github.com/mattermost/mattermost-server/issues/19908

#### Checklist

~~- Telemetry updated~~
~~- Gated by experimental feature flag~~
- [X] Unit tests updated

### CODE REVIEW

**Comment by lieut-data** [comment]
Sweet, thank you @jeffreyssmith2nd! I've asked two of our core committers to help review :)

**Comment by jeffreyssmith2nd** [comment]
Hey @agarciamontoro, thanks for the feedback.

I noticed there weren't any tests around the `/reminder` route and I wasn't sure if that was intentional, sounds like it was just yet to be implemented. I'd be happy to add those tests, I'll look into it probably early next week and get this PR updated with them.

**Comment by agarciamontoro** [comment]
> sounds like it was just yet to be implemented.

Yup, that was it!

> I'd be happy to add those tests, I'll look into it probably early next week and get this PR updated with them.

Great, thank you! Let us know if you have any question :)

File: server/app/reminder.go [hunk 1]
```diff
@@ -162,6 +162,28 @@ func (s *PlaybookRunServiceImpl) resetReminderTimer(playbookRunID string) error
 	return nil
 }
 
+func (s *PlaybookRunServiceImpl) ResetReminder(playbookRunID string, newReminder time.Duration) error {
```
Comment (trilopin) [inline server/app/reminder.go:165]: it would be great if you add a comment describing this public func.

**Review by trilopin — COMMENTED** [review]
Thank you for your help!

I've just added a tiny request.

**Comment by lieut-data** [comment]
@jeffreyssmith2nd, can we help with the changes described above?

**Comment by jeffreyssmith2nd** [comment]
Hey @lieut-data sorry for the delay here, got busy with other stuff. I did have a chance to look at this last night and started making progress on it.

**Comment by lieut-data** [comment]
Awesome, thank you @jeffreyssmith2nd!

File: client/reminders.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+type RemindersService struct {
+	client *Client
+}
+
+func (s *RemindersService) Reset(ctx context.Context, playbookRunID string, payload ReminderResetPayload) error {
+	resetURL := fmt.Sprintf("runs/%s/reminder", playbookRunID)
+
+	fmt.Println(s)
+	fmt.Println(s.client)
```
Comment (trilopin) [inline client/reminders.go:18]: I guess we can remove these prints

```suggestion
```

Comment (jeffreyssmith2nd) [reply client/reminders.go]: Fixed, sorry should have caught that

**Review by agarciamontoro — APPROVED** [review]
This looks great, @jeffreyssmith2nd, thank you!

**Comment by agarciamontoro** [comment]
I saw that there's a failure in the linter, but it is not from a file in this PR :thinking: I'll merge master into your branch to see if this fixes it.

**Comment by jeffreyssmith2nd** [comment]
> Ugh, that lint error was fixed in the very latest commit. Updating again. @jeffreyssmith2nd, ~~could you also run `make mocks` again to solve the conflict?~~
> 
> EDIT: The automatic `/update-branch` did not work, of course, so I took the liberty to do it manually and push the changes along with the `make mocks`.

Great, thanks for fixing that!

**Review by Phrynobatrachus — CHANGES_REQUESTED** [review]
This looks pretty good so far, but I think there's still a few things left to handle. In the timeline view, the new events look like this:

Screenshot from 2022-05-16 12-33-17

These should be filled out. Here's some existing ones for comparison:

Screenshot from 2022-05-16 13-21-00

I think the flag icon would be appropriate for this event as well. There is also a filter for different types of timeline events:

Screenshot from 2022-05-16 13-08-59

Currently, these events only show with `All events` checked. To me it would make sense to include these with the `Status updates` checkbox, but 0/5 on that.

So in total, this is what I believe needs to be added.

- [ ] fields on the `TimelineEvent` in `server/app/reminder.go`, probably `Summary` and/or `Details`
- [ ] case handler for the event in the webapp component `webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/timeline_event_item.tsx`
- [ ] filter handling in `webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/timeline_retro.tsx`

cc @agarciamontoro regarding how to include this event in the filters

**Comment by agarciamontoro** [comment]
@jeffreyssmith2nd, I took a look at this and the missing info seems to be rendered directly by [the webapp code](https://github.com/mattermost/mattermost-plugin-playbooks/blob/master/webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/timeline_event_item.tsx#L195-L265). You only need to add another case to that switch to handle the new event you added.

By the way, the conflicting files here will be solved by merging master and running `make mocks`. Let us know if there's anything else we can help with!

**Comment by Phrynobatrachus** [comment]
Closing this in favor of #1350. :tada:

**Pull request closed without merging**